### PR TITLE
Special characters plugin

### PR DESCRIPTION
**Pull request by mlewand**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Introduced the special characters plugin. Closes ckeditor/ckeditor5#1110.

---

### Additional information

**You can use the [constellation branch](https://github.com/ckeditor/ckeditor5/pull/5818) for easier setup.**

Subpr:

* ckeditor/ckeditor5-theme-lark#256

This PR has:

* Basic public API
* Basic API docs
* Proposed set of categories
* UI
* `InsertSpecialCharacterCommand` state is integrated with `input` command
* Test coverage

Missing:

* The "all" category
* API docs for characters plugin like `SpecialCharactersMathematical`
* Feature guide
* Translations for labels (we decided to extract that as a followup)
* I'd like to take a closer look at data structure in SpecialCharacter plugin

### CODE REVIEW

File: src/specialcharactersui.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	 */
+	static get pluginName() {
+		return 'SpecialCharactersUI';
+	}
+
+	init() {
+		const editor = this.editor;
+		const t = editor.t;
+		const specialCharsPlugin = editor.plugins.get( 'SpecialCharacters' );
+		const command = new InsertSpecialCharacterCommand( editor );
```
Comment (pomek) [inline src/specialcharactersui.js]:
```suggestion
		const command = new InsertSpecialCharacterCommand( editor );
		const command = editor.commands.get( 'insertSpecialCharacter' );
```

and:

```js
editor.commands.add( 'specialCharacters', command );
```

should be removed.

The command is being registered in the editing part of the feature now.

Comment (mlewand) [reply src/specialcharactersui.js]: Thnx @pomek addressed this in bc5f58928401b590659ce629ccd4e199f6b12ccd

File: src/specialcharacters.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+		return this._characters.get( title );
+	}
+
+	/**
+	 * Returns a group of special characters. If the group with the specified name does not exist, it will be created.
+	 *
+	 * @param {String} groupName A name of group to create.
+	 */
+	_getGroup( groupName ) {
+		/* istanbul ignore else */
```
[the remark this answers is not in the document]
Comment (mlewand) [reply src/specialcharacters.js]: A todo that was not fixed in time, fix landed on master.

File: theme/charactergrid.css [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/*
```
[the remark this answers is not in the document]
Comment (oleq) [reply theme/charactergrid.css]: [Media embed does something similar](https://github.com/ckeditor/ckeditor5-media-embed/tree/master/theme). I'm cool with it as long as the component has enough complexity to have its own stylesheet and I suppose the char grid qualifies.

**Pull request merged**